tls_manager.go

```diff
@@ -123,9 +185,77 @@ func (t *TLSManager) createTLSPair() error {
 		if err != nil {
 			return err
 		}
+
+		if t.cfg.TLSEncryptKey {
+			var b bytes.Buffer
+			e, err := lnencrypt.KeyRingEncrypter(keyRing)
+			if err != nil {
+				return fmt.Errorf("unable to create "+
+					"encrypt key %v", err)
+			}
+
+			err = e.EncryptPayloadToWriter(
+				keyBytes, &b,
+			)
+			if err != nil {
+				return err
+			}
+
+			keyBytes = b.Bytes()
+		}
+
+		err = cert.WriteCertPair(
```

Comment:
Since we now wrote the encrypted key and cert, can't we return early now? Otherwise we try to re-encrypt it below with `t.encryptPlaintextKey()`.

Reply:
Good call